projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py

```diff
@@ -177,10 +205,22 @@ def initialize_job(self, context: JobContext) -> None:
 
         root_logger.addHandler(handler)
 
+        syslog_enabled = context.core_context.configuration.get_value(SYSLOG_ENABLED_KEY)
+        syslog_host = context.core_context.configuration.get_value(SYSLOG_HOST_KEY)
+        syslog_port = context.core_context.configuration.get_value(SYSLOG_PORT_KEY)
+        syslog_protocol = context.core_context.configuration.get_value(SYSLOG_PROTOCOL_KEY)
+        attempt_id = context.core_context.state.get(CommonStoreKeys.ATTEMPT_ID)
+        syslog_handler = configure_syslog_handler(syslog_enabled, syslog_host, syslog_port,
+                                                  syslog_protocol, job_name, attempt_id)
+        if syslog_handler:
+            root_logger.addHandler(syslog_handler)
+
         out: HookCallResult
         out = yield
 
         root_logger.removeHandler(handler)
+        if syslog_handler:
```

Comment:
We shouldn't be removing the syslog handler, we need it til the end of the job run. I had a problem previously, where adding a handler in this hook didn't result in having the handler in the next hook. That's why we remove it here and add it again in the `run_job` hook.

So we should either not remove the handler here or remove it here and add it again in the next hook. This should be tested in debug mode.